frontend/src/app/lib/install-addon.js

```diff
@@ -3,19 +3,52 @@ import cookies from 'js-cookie';
 import EmailDialogView from '../views/email-opt-in-dialog-view';
 
 
+const MOZADDONMANAGER_ALLOWED_HOSTNAMES = [
+  'testpilot.firefox.com',
+  'testpilot.stage.mozaws.net',
+  'testpilot.dev.mozaws.net'
+];
+
+
+function mozAddonManagerInstall(url) {
+  return navigator.mozAddonManager.createInstall({ url }).then(install => {
+    return new Promise((resolve, reject) => {
+      install.addEventListener('onInstallEnded', () => resolve());
+      install.addEventListener('onInstallFailed', () => reject());
+      install.install();
+    });
+  });
+}
+
 function installAddon(view, eventCategory, callback) {
```

Comment:
@chuckharmston I think if you pass in `navigator` here, you'll easily be able to mock out `mozAddonManager` for testing.

It [looks like](https://github.com/mozilla/testpilot/blob/master/frontend/src/test/browser.js#L8) the existing frontend tests expect a test/lib directory to exist, so you should be able to just copy an existing test into a new lib directory and add the happy path test.

Happy to help further if you'd like; let me know